pepper-apis/src/main/java/org/broadinstitute/ddp/content/SelectedPickListOptionRenderUtil.java

```diff
@@ -0,0 +1,67 @@
+package org.broadinstitute.ddp.content;
+
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.broadinstitute.ddp.model.activity.definition.question.PicklistOptionDef;
+import org.broadinstitute.ddp.model.activity.definition.question.PicklistQuestionDef;
+import org.broadinstitute.ddp.model.activity.definition.question.QuestionDef;
+import org.broadinstitute.ddp.model.activity.instance.answer.Answer;
+import org.broadinstitute.ddp.model.activity.instance.answer.PicklistAnswer;
+import org.broadinstitute.ddp.model.activity.instance.answer.SelectedPicklistOption;
+import org.broadinstitute.ddp.model.activity.instance.question.PicklistOption;
+import org.broadinstitute.ddp.model.activity.instance.question.PicklistQuestion;
+import org.broadinstitute.ddp.model.activity.instance.question.Question;
+
+
+/**
+ * Utility methods for rendering selected picklist options and picklist option detail text.
+ */
+public class SelectedPickListOptionRenderUtil {
+
+    public static String selectedOptionLabelsRender(QuestionDef questionDef, Answer answer, String fallbackValue, String isoLangCode) {
+        Map<String, PicklistOptionDef> options = ((PicklistQuestionDef) questionDef)
+                .getAllPicklistOptions().stream()
+                .collect(Collectors.toMap(PicklistOptionDef::getStableId, Function.identity()));
+        return ((PicklistAnswer) answer).getValue().stream()
+                .map(selected -> renderOptionLabelOrFallback(options, selected, fallbackValue, isoLangCode))
+                .collect(Collectors.joining(","));
+    }
+
+    public static String selectedOptionLabelsRender(Question question, Answer answer, String fallbackValue) {
+        Map<String, String> options = ((PicklistQuestion) question)
+                .streamAllPicklistOptions()
+                .collect(Collectors.toMap(PicklistOption::getStableId,
+                        (p) -> p.getOptionLabel() == null ? fallbackValue : p.getOptionLabel()));
+        return ((PicklistAnswer) answer).getValue().stream()
+                .map(selected -> options.get(selected.getStableId()))
+                .collect(Collectors.joining(","));
+    }
+
+    public static String detailTextRender(QuestionDef questionDef, Answer answer, String fallbackValue, String isoLangCode) {
+        String detailText = getDetailText(answer);
```

Comment:
Hmmm, recall that EACH picklist option MAY allow inputting additional details. In our use-case for PanCan, there's only one option with detail text, but that might not be the case for others. Notice how in the methods above we're doing "map" and "collect" and joining with a semicolon. I think what we should do here, is to iterate through all the selected options (like what we're doing in `selectedOptionLabelsRender()`) and if `useDetails` is true AND the selected option has detail text, then we use that instead of the option label. So I think it might be better off passing `useDetails` into `selectedOptionLabelsRender()` and handling it there.

Reply:
Fixed